mempool/src/interface/mock.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(clippy::unwrap_used)]
+
+use crate::error::Error;
+use crate::tx_accumulator::TransactionAccumulator;
+use crate::MempoolEvent;
+use common::chain::{SignedTransaction, Transaction};
+use common::primitives::Id;
+use std::sync::{Arc, Mutex};
+use subsystem::{CallRequest, ShutdownRequest};
+
+use super::mempool_interface::{MempoolInterface, MempoolSubsystemInterface};
+
+#[derive(Clone)]
+pub struct MempoolInterfaceMock {
```

Comment:
could it be done using `mockall` instead?

Reply:
I tried. Although `mockall` says that it supports `async`, no matter what I tried, I couldn't get it to work with our Subsystem